python/anvill/binja.py

```diff
@@ -173,199 +173,260 @@ def _convert_bn_llil_type(
+        return MacOS()
+    elif "windows" in platform:
+        return WindowsOS()
+    else:
+        raise UnhandledOSException(
+            "Missing operating system object type for OS '{}'".format(platform)
+        )
+
 def _cache_key(tinfo: bn.types.Type) -> str:
     """ Convert bn Type instance to cache key"""
     return str(tinfo)
 
-def _convert_named_type_reference(
-    bv, tinfo: bn.types.Type, cache) -> Type:
-    """ Convert named type references into a `Type` instance"""
-    if tinfo.type_class != bn.TypeClass.NamedTypeReferenceClass:
-        return
+class TypeCache(object):
+    def __init__(self, bv):
+        self._bv = bv
+        self._cache = dict()
+
+    def __convert_struct(self, tinfo: bn.types.Type):
+        """Convert bn struct type into a `Type` instance"""
+
+        if (tinfo.structure.type !=
+             bn.StructureType.StructStructureType):
+            return
 
-    named_tinfo = tinfo.named_type_reference
-    if (named_tinfo.type_class
-        == bn.NamedTypeReferenceClass.StructNamedTypeClass):
-        # Get the bn struct type and recursively recover the elements
-        ref_type = bv.get_type_by_name(named_tinfo.name);
-        struct_type =  ref_type.structure
         ret = StructureType()
-        cache[_cache_key(struct_type)] = ret
-        for elem in struct_type.members:
-            ret.add_element_type(_convert_bn_type(bv, elem.type, cache))
+        self._cache[_cache_key(tinfo)] = ret
+        for elem in tinfo.structure.members:
+            ret.add_element_type(self._convert_bn_type(elem.type))
+
         return ret
 
-    elif (named_tinfo.type_class
-        == bn.NamedTypeReferenceClass.UnionNamedTypeClass):
-        # Get the union type and recover the member elements
-        ref_type = bv.get_type_by_name(named_tinfo.name);
-        struct_type =  ref_type.structure
+    def __convert_union(self, tinfo: bn.types.Type):
+        """Convert bn union type into a `Type` instance"""
+
+        if (tinfo.structure.type !=
+             bn.StructureType.UnionStructureType):
+            return
+
         ret = UnionType()
-        cache[_cache_key(struct_type)] = ret
-        for elem in struct_type.union.members:
+        cache[_cache_key(tinfo)] = ret
+        for elem in tinfo.structure.union.members:
             ret.add_element_type(_convert_bn_type(bv, elem.type, cache))
-        return ret
 
-    elif (named_tinfo.type_class
-        == bn.NamedTypeReferenceClass.TypedefNamedTypeClass):
-        ref_type = bv.get_type_by_name(named_tinfo.name);
-        ret = TypedefType()
-        cache[_cache_key(ref_type)] = ret
-        ret.set_underlying_type(_convert_bn_type(bv, ref_type, cache))
         return ret
 
-    elif (named_tinfo.type_class
-        == bn.NamedTypeReferenceClass.EnumNamedTypeClass):
-        # Set the underlying type int of size width
-        ref_type = bv.get_type_by_name(named_tinfo.name);
+    def __convert_enum(self, tinfo: bn.types.Type) -> Type:
```

Comment:
I think we should decide if we want to use type hints in the code or not. I have a slight preference for having them, but I don't really mind either way. What's important to me is consistency.